File: miklashevsky2017_LDT_RussianNouns/generate_prompts.R
library(dplyr)
library(readr)
library(jsonlite)

## read data
data = read_csv("processed_data/exp1.csv", show_col_types = FALSE)

all_prompts = list()

participants_list = unique(data$participant_id)
trials_exp1 = 0:max(data$trial_order, na.rm = TRUE)

instruction_practice_mapping1 =
  paste0(
    'Здравствуйте!\n',
    'Спасибо, что согласились принять участие в эксперименте.\n',
    'Сейчас перед Вами будут появляться стимулы - слова и псевдослова (последовательности букв, поxожие на слова). Ваша задача - нажать на нужную клавишу:\n',
    '1 - если перед Вами настоящее слово\n',
    '2 - если перед Вами не настоящее слово (псевдослово).\n',
    'Сначала будет небольшая тренировка, чтобы Вы могли лучше понять задание.\n',
    'Нажмите пробел, чтобы продолжить.\n'
  )

instruction_main_experiment_mapping1 =
  paste0(
    'Поxоже, что Вы уже освоились. Сейчас начнется настоящий эксперимент.\n',
    'Помните, Ваша задача - нажать на нужную клавишу:\n',
    '1 - если перед Вами настоящее слово\n',
    '2 - если перед Вами не настоящее слово (псевдослово).\n',
    'Готовы?\n',
    'Нажмите пробел, чтобы продолжить.\n'
  )

instruction_practice_mapping2 =
  paste0(
    'Здравствуйте!\n',
    'Спасибо, что согласились принять участие в эксперименте.\n',
    'Сейчас перед Вами будут появляться стимулы - слова и псевдослова (последовательности букв, поxожие на слова). Ваша задача - нажать на нужную клавишу:\n',
    '2 - если перед Вами настоящее слово\n',
    '1 - если перед Вами не настоящее слово (псевдослово).\n',
    'Сначала будет небольшая тренировка, чтобы Вы могли лучше понять задание.\n',
    'Нажмите пробел, чтобы продолжить.\n'
  )

instruction_main_experiment_mapping2 =
  paste0(
    'Поxоже, что Вы уже освоились. Сейчас начнется настоящий эксперимент.\n',
    'Помните, Ваша задача - нажать на нужную клавишу:\n',
    '2 - если перед Вами настоящее слово\n',
    '1 - если перед Вами не настоящее слово (псевдослово).\n',
    'Готовы?\n',
    'Нажмите пробел, чтобы продолжить.\n'
  )

# participant-level info (all columns that are constant within participant)
participant_info =
  data %>%
  group_by(participant_id) %>%
  summarise(
    response_mapping   = first(response_mapping),
    age                = first(age),
    first_language     = first(first_language),
    gender             = first(gender),
    education_subject  = first(education_subject),
    list               = first(list),
    .groups = "drop"
  )

for (ppt in participants_list) {
  
  exp_participant = data %>% filter(participant_id == ppt)
  info = participant_info %>% filter(participant_id == ppt)
  
  mapping = info$response_mapping[1]
  
  if (mapping == "DB_LD_1") {
    instruction_practice = instruction_practice_mapping1
    instruction_main = instruction_main_experiment_mapping1
  } else {
    instruction_practice = instruction_practice_mapping2
    instruction_main = instruction_main_experiment_mapping2
  }
  
  individual_prompt = paste0(instruction_practice, "\n", instruction_main, "\n")
  
  for (trial in trials_exp1) {
    exp_trial = exp_participant %>% filter(trial_order == trial)
    if (nrow(exp_trial) > 0) {
      
      stimulus  = exp_trial$target_word[1]
      response  = exp_trial$response[1]
      correct   = exp_trial$correct_response[1]
      rt        = exp_trial$rt[1]
      condition = exp_trial$condition[1]
      acc       = exp_trial$accuracy[1]
      block     = exp_trial$BlockList[1]
      
      datapoint =
        paste0(
          "trial ", trial, ": ",
          stimulus, " (", condition, "). ",
          "Correct response: ", correct,
          ". You press <<", response, ">>",
          ". ACC=", acc,
          ". RT=", rt, "ms.\n"
        )
      
      individual_prompt = paste0(individual_prompt, datapoint)
    }
  }
  
  all_prompts[[length(all_prompts) + 1]] = list(
    text = individual_prompt,
    experiment = "myexperiment2017_LDT",
    participant = ppt,
    participant_info = as.list(info)
  )
}

# Save all prompts to JSONL file
con = file("prompts.jsonl", open = "w", encoding = "UTF-8")
for (i in seq_along(all_prompts)) {
  writeLines(toJSON(all_prompts[[i]], auto_unbox = TRUE, null = "null"), con)
}
close(con)

File: wang2025_lexicaldecision/generate_prompts.py
# Generate prompts for wang2025_lexicaldecision

# Load libararies
import pandas as pd
import jsonlines

# Load data
df = pd.read_csv("/Users/cyhsieh/PsychLing-101/wang2025_lexicaldecision/processed_data/exp1.csv")

# create empty list to store all prompts
all_prompts = []

###########################
# Megastudy #
###########################

participant_list = df['participant_id'].unique()
trial_num = range(df['trial_id'].max() + 1)

# Instructions

instruction = "每次试验包括一个注视标记和一个刺激（真字或假字）。首先，屏幕中央呈现注视标记500毫秒，120毫秒后呈现刺激。如果你认为该刺激是真字，请按j键；如果认为是假字，请按f键。虽然没有反应时间限制，但请尽可能快速并尽最大准确性作答。如果你曾经见过或认识这些刺激，那么它很有可能是真字；如果你没有见过，那么它很可能不是真字。"

# Generate individual prompts for participants
for participant in participant_list:
    exp_participant = df[df['participant_id'] == participant]
    #age = exp_participant['age'].iloc[0].item()
    individual_prompt = instruction
    for trial in trial_num:
        exp_trial = exp_participant.loc[exp_participant['trial_id'] == trial]
        if not exp_trial.empty:  # Only process if trial exists for this participant
            image = exp_trial['image_filename'].iloc[0]
            response = exp_trial['response'].iloc[0]
            #trial_instruction = exp_trial['trial_instruction'].iloc[0]
            trial_id = exp_trial['trial_id'].iloc[0]
            accuracy = exp_trial['accuracy'].iloc[0]
            rt = exp_trial['rt'].iloc[0]
            datapoint = f'试验{trial_id}：{image}。你按下了<<{response}>>键。{accuracy}。反应时间为 {rt} 毫秒。\n'
            individual_prompt += datapoint
    all_prompts.append({'text': individual_prompt, 'experiment': 'wang2025_lexicaldecision', 'participant_id': participant, 'rt': rt})

# Save all prompts to JSONL file
with jsonlines.open("prompts.jsonl", "w") as writer:
    writer.write_all(all_prompts)
